### PR TITLE
[WINDUPRULE-838] Spring cloud detection

### PR DESCRIPTION
**Pull request by jmle**
See [https://issues.redhat.com/browse/WINDUPRULE-838](https://issues.redhat.com/browse/WINDUPRULE-838).

### CODE REVIEW

**Comment by PhilipCattanach** [comment]
@jmle - Thank you. The rule looks perfect but there is a problem with the test data file. We don'rt actually embed the true spring artifact in our data folder. Rather we use a trick to keep the size of the folder down.
https://github.com/windup/windup-rulesets/blob/1d94433348a11243b77897a99ae0f6a4f1305915/rules-reviewed/technology-usage/tests/.mta/cache/nexus-indexer-data/test.archive-metadata.txt

File: rules-reviewed/technology-usage/tests/.mta/cache/nexus-indexer-data/test.archive-metadata.txt [hunk 1]
```diff
@@ -1,2 +1,3 @@
 2d6b521afd7410f3b03640a7a109935d9ebf8025 io.micrometer:micrometer-core:1.3.0
 9a15a7c84bd12516574bcaf87ffa38c1e65e8a2b org.springframework:spring-beans:5.1.3.RELEASE
+52c4beb42aa8637e27ed1c60e65daee606d2e6f5 org.springframework:spring-cloud-function-context
```
Comment (mrizzi) [inline rules-reviewed/technology-usage/tests/.mta/cache/nexus-indexer-data/test.archive-metadata.txt:3]:
```suggestion
52c4beb42aa8637e27ed1c60e65daee606d2e6f5 org.springframework.cloud:spring-cloud-function-context
```

**Review by mrizzi — COMMENTED** [review]
I've changed the provided `groupId` to match the expected one.

**Comment by PhilipCattanach** [comment]
Tested successfully - I will merge to master as soon as the automated tests complete.

**Pull request merged**